### PR TITLE
docs: Add example with object pattern to computed-property-spacing

### PR DESCRIPTION
**Pull request by whizzzkid**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

- [x] Documentation update
- [ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
- [ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
- [ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
- [ ] Add autofixing to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)
I faced https://github.com/eslint/eslint/issues/14002 and added an example and wording to help others find it.

#### Is there anything you'd like reviewers to focus on?
the wording and the right place.

### CODE REVIEW

File: docs/rules/computed-property-spacing.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ var a = "prop";
 var obj = {
   [a]: "value" // computed property key in object literal (ECMAScript 6)
 };
+
+// applies to the spacing for dynamic keys (computed property key) when destructuring objects.
+const ({ [a]: someProp } = obj); // someProp now maps to the value of prop.
```
Comment (btmills) [inline docs/rules/computed-property-spacing.md:19]: Parentheses are invalid syntax here:

```suggestion
const { [a]: someProp } = obj; // someProp now maps to the value of prop.
```

**Review by btmills — CHANGES_REQUESTED** [review]
Thanks, documenting the destructuring case is helpful! I left an inline suggestion to fix a syntax error in the example, but this'll be good to go once that's fixed.

File: docs/rules/computed-property-spacing.md (hunk 1 above)
Comment (whizzzkid) [reply docs/rules/computed-property-spacing.md]: Wait, I think I did not read this right, that syntax looks wrong, that works, when assigning to a delcared variable:

```js
let someProp;
({ [a]: someProp } = obj);
```

but if we are declaring a new variable:

```js
const { [a]: someProp } = obj;
```

this is mentioned in this section: [Assignment without declaration](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

**Comment by mdjermanovic** [comment]
I marked this as 'evaluating' for now since there's an issue related to this change https://github.com/eslint/eslint/issues/12048 and it isn't accepted yet.

I think it's most likely that we will accept this documentation update, at least for the `computed-property-spacing` rule, but we should also add some test cases to confirm this behavior. It can be a separate PR.

**Comment by mdjermanovic** [comment]
Closing due to age.

**Pull request closed without merging**